binding/web/README.md

```diff
@@ -1,20 +1,316 @@
+
+## Compatibility
+
+- Chrome / Edge
+- Firefox
+- Safari
+
+## Installation
+
+### Package
+
+Using `Yarn`:
+
+```console
+yarn add @picovoice/rhino-web
+```
+
+or using `npm`:
+
+```console
+npm install --save @picovoice/rhino-web
+```
+
+### AccessKey
+
+Rhino requires a valid Picovoice `AccessKey` at initialization. `AccessKey` acts as your credentials when using
+Rhino SDKs.
+You can get your `AccessKey` for free. Make sure to keep your `AccessKey` secret.
+Signup or Login to [Picovoice Console](https://console.picovoice.ai/) to get your `AccessKey`.
+
+## Usage
+
+There are two methods to initialize Rhino:
+
+### Public Directory
+
+**NOTE**: Due to modern browser limitations of using a file URL, this method does __not__ work if used without hosting a server.
+
+This method fetches [the model file](https://github.com/Picovoice/rhino/blob/master/lib/common/rhino_params.pv) from the public directory and feeds it to Rhino. Copy the model file into the public directory:
+
+```console
+cp ${RHINO_MODEL_FILE} ${PATH_TO_PUBLIC_DIRECTORY}
 ```
-yarn
-yarn build
+
+### Base64
+
+**NOTE**: This method works without hosting a server, but increases the size of the model file roughly by 33%.
+
+This method uses a base64 string of the model file and feeds it to Rhino. Use the built-in script `pvbase64` to base64 your model file:
+
+```console
+npx pvbase64 -i ${RHINO_MODEL_FILE} -o ${OUTPUT_DIRECTORY}/${MODEL_NAME}.js
+```
+
+The output will be a js file which you can import into any file of your project. For detailed information about `pvbase64`,
+run:
+
+```console
+npx pvbase64 -h
 ```
 
-Now each individual project will exist. E.g.:
+### Init options
+
+Rhino saves and caches your model file in IndexedDB to be used by Web Assembly. Use a different `customWritePath`
+variable to hold multiple model values and set the `forceWrite` value to true to force re-save the model file.
+Set `processErrorCallback` to handle errors if an error occurs while transcribing.
+If the model file (`.pv`) changes, `version` should be incremented to force the cached model to be updated.
 
+```typescript
+// these are default
+const options = {
+  sensitivity: 0.5,
+  endpointDurationSec: 1.0,
+  requireEndpoint: true,
+  processErrorCallback: (error) => {},
+  customWritePath: "rhino_model",
+  forceWrite: false,
+  version: 1,
+}
 ```
-cd rhino-web-en-worker
-yarn
-yarn build
+
+### Initialize in Main Thread
+
+Create a `inferenceCallback` function to get the results from the engine:
+
+```typescript
+function inferenceCallback(inference) {
+  if (inference.isFinalized) {
+    if (inference.isUnderstood) {
+      console.log(inference.intent)
+      console.log(inference.slots)
+    }
+  }
+}
 ```
+
+Add to the `options` object an `processErrorCallback` function if you would like to catch errors:
+
+```typescript
+function processErrorCallback(error: string) {
+...
+}
+options.processErrorCallback = processErrorCallback;
+```
+
```

Comment:
I don't think `processErrorCallback` is used in non-worker is it?

Reply:
okay it looks like it is now, you just need to add it to the doc comment above the constructor so that folks know it's optional in mainthread